frontend/src/routes/Governance/discovered/ByCluster/common.tsx

```diff
@@ -0,0 +1,217 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { useMemo } from 'react'
+import { ViolationsCard, ViolationSummary } from '../../overview/PolicyViolationSummary'
+import { DiscoveredPolicyItem, DiscoverdPolicyTableItem, ISourceType } from '../useFetchPolicies'
+import { compareStrings, IAcmTableColumn } from '../../../../ui-components'
+import { TFunction } from 'react-i18next'
+import { getPolicySource, parseStringMap } from '../../common/util'
+import { generatePath, Link } from 'react-router-dom-v5-compat'
+import { NavigationPath } from '../../../../NavigationPath'
+import { Channel, HelmRelease, Subscription } from '../../../../resources'
+import { CheckCircleIcon, ExclamationCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons'
+import { useTranslation } from '../../../../lib/acm-i18next'
+
+export const policyViolationSummary = (discoveredPolicyItems: DiscoveredPolicyItem[]): ViolationSummary => {
+  let compliant = 0
+  let noncompliant = 0
+  let pending = 0
+  let unknown = 0
+  for (const policy of discoveredPolicyItems) {
+    if (policy.disabled || !policy?.compliant) continue
+    switch (policy?.compliant?.toLowerCase()) {
+      case 'compliant':
+        compliant++
+        break
+      case 'noncompliant':
+        noncompliant++
+        break
+      case 'pending':
+        pending++
+        break
+      default:
+        unknown++
+        break
+    }
+  }
+  return { noncompliant, compliant, pending, unknown }
+}
+
+export const ByClusterCols = (
+  t: TFunction<string, undefined>,
+  helmReleases: HelmRelease[],
+  subscriptions: Subscription[],
+  channels: Channel[],
+  moreCols?: IAcmTableColumn<DiscoveredPolicyItem>[]
+): IAcmTableColumn<DiscoveredPolicyItem>[] =>
+  useMemo(() => {
+    return [
+      {
+        header: t('Cluster'),
+        cell: (item: DiscoveredPolicyItem) => {
+          const labelVal = parseStringMap(item.label)['policy.open-cluster-management.io/policy']
+          // Parent Policy exist
+          if (labelVal?.split('.').length == 2) {
+            return (
+              <Link
+                to={generatePath(NavigationPath.policyTemplateDetails, {
+                  namespace: labelVal.split('.')[0],
+                  // `kind: Policy` name
+                  name: labelVal.split('.')[1],
+                  clusterName: item.cluster,
+                  apiVersion: item.apiversion,
+                  apiGroup: item.apigroup,
+                  kind: item.kind,
+                  // discovered policy name
+                  templateName: item.name,
+                })}
+              >
+                {item.cluster}
+              </Link>
+            )
+          }
+          return (
```

Comment:
Could we just always return this? I think the fact that it was deployed by a `Policy` shouldn't change the navigation experience here.

Reply:
always link to ` generatePath(NavigationPath.discoveredPolicyDetails` ?

Reply:
policyTemplateDetails head nav link is different.  `policyTemplateDetails` has  `kind:Policy` link

Reply:
@yiraeChristineKim I think the breadcrumbs should be consistent. So if you reached the policy template details from the "Policies" table, then it should show the "Policies" table breadcrumbs. If you reached the policy template details for the same ConfigurationPolicy from the "Discovered policies" table, then it should show the "Discovered policies" table breadcrumbs.

Consistency is better in my opinion.